recipes/matio/all/conanfile.py

```diff
@@ -0,0 +1,102 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MatioConan(ConanFile):
+    name = "matio"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/matio/"
+    description = "Matio is a C library for reading and writing binary MATLAB MAT files."
+    topics = ("conan", "matlab", "mat-file", "file-format", "hdf5")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/*"
+    options = {
+        "shared": [True, False],
+        "extended_sparse": [True, False],
+        "fPIC": [True, False],
+        "mat73": [True, False],
+        "with_hdf5": [None, "shared", "static"],
+        "with_zlib": [None, "shared", "static"]
+    }
+    default_options = {
+        "shared": False,
+        "extended_sparse": True,
+        "fPIC": True,
+        "mat73": True,
+        "with_hdf5": "static",
+        "with_zlib": "static"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def requirements(self):
+        if self.options.get_safe("with_hdf5") not in (None, "None"):
+            self.requires("hdf5/[>=1.8 <1.13]")
+        if self.options.get_safe("with_zlib") not in (None, "None"):
+            self.requires("zlib/[>=1.2.3]")
```

Comment:
Just use the latest cci versions.
```suggestion
        if self.options.with_hdf5:
            self.requires("hdf5/1.12.0")
        if self.options.get_safe("with_zlib") not in (None, "None"):
            self.requires("zlib/1.2.11")
```

Reply:
Hm , there will be hdf5/1.12.1 any time soon and my intention was to accept that one, too.

Reply:
Consumers can always override the version in their build tree.
The use of None is currently failing CI.

Reply:
Is the range acceptable at all?

Reply:
It's not used by any other recipe.
The "others" will also make the same remark.
We would have to update this recipe anyhow when hdf5/1.13 is released.

Reply:
> It's not used by any other recipe.

OK, going to change it.

> We would have to update this recipe anyhow when hdf5/1.13 is released.

Well, also the matio library is not ready for hdf5 > 1.12 because of there interface changes.